src/olympia/lib/crypto/tasks.py

```diff
@@ -50,20 +50,18 @@
 add-on, please sign in to addons.mozilla.org and delete your add-on(s).
 """  # noqa: E501
 
-version_regex = re.compile(
-    r'^(?P<prefix>.*)(?P<version>\.1\-signed)(|\-(?P<number>\d+))$'
-)
-
 
 def get_new_version_number(version):
-    match = version_regex.search(version)
-    if not match:
-        return f'{version}.1-signed'
-    else:
-        num = int(match.groupdict()['number'] or 1)
-        return '{}{}-{}'.format(
-            match.groupdict()['prefix'], match.groupdict()['version'], num + 1
-        )
+    # Parse existing version number, increment the last part, and replace
+    # the suffix with "resigned1", in order to get a new version number that
+    # would force existing users to upgrade.
+    # Example: '1.0' would return '1.1resigned1'.
+    vs = VersionString(version)
+    parts = vs.vparts
+    parts[-1].a += 1
+    parts[-1].b = 'resigned'
+    parts[-1].c = 1
+    return str(VersionString.from_vparts(parts))
```

Comment:
@eviljeff you're the resident `VersionCompare` expert. Does this (and the couple new methods I had to add to `VersionCompare` to make that work) makes sense to you?

Reply:
It _should_ produce a valid increased version string, but `vparts` isn't guaranteed to store all the same information that is in the underlying version string - it's there only for comparison between two strings.  I.e. there is no certainty that `VersionString.from_vparts(VersionString('foo').vparts) == VersionString('foo')` so the new string after it's been adjusted may not "look like" the existing string.  Maybe that's fine, though.

Reply:
(btw you don't need to cast `VersionString` back to a `str` - it subclasses `str`)

Reply:
mmm, I guess it would be cleaner to live in `get_new_version_string`.  It wouldn't need a function imo, just something like:

```suggestion
    vs = VersionString(version)
    parts = vs.vparts
    parts[-1].a += 1
    parts[-1].b = 'resigned'
    parts[-1].c = 1
    return VersionString('.'.join(str(part) for part in parts)
```

(with the `__str__` suggestion below)